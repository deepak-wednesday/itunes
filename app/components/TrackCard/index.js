/**
 *
 * TrackCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card } from 'antd';
import { PlayCircleTwoTone, RightCircleTwoTone, PauseCircleTwoTone } from '@ant-design/icons';
import { Link } from 'react-router-dom';
import * as colors from '@app/themes/colors';
import If from '@components/If';

const { Meta } = Card;

const CustomCard = styled(Card)`
  && {
    width: 18em;
    margin: 1rem 0;
    height: ${(props) => (props.height ? props.height : 30)}em;
    background-color: ${colors.trackCardColor};
  }
`;
const ImageCard = styled(Card)`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 15em;
  height: ${(props) => (props.height ? props.height : 24)}em;
`;
const TextCard = styled(Meta)`
  && {
    margin: 0.3rem !important;
    width: 15em;
    padding: 1em;
  }
`;
const Image = styled.img`
  width: 15em;
  height: ${(props) => (props.height ? props.height : 18)}em;
`;
const CustomPause = styled(PauseCircleTwoTone)`
  font-size: 2rem;
  padding: 1rem;
  cursor: pointer;
`;
const CustomPlay = styled(PlayCircleTwoTone)`
  font-size: 2rem;
  padding: 1rem;
  cursor: pointer;
`;
const CustomRight = styled(RightCircleTwoTone)`
  font-size: 2rem;
  padding: 1rem;
  cursor: pointer;
`;
const CustomButton = styled.div`
  dispaly: flex;
  justify-content: center;
  align-items: center;
`;

export function TrackCard({ item, currentPlayingId, isPlaying, onPlay, onPause }) {
  const { trackName, artistName, artworkUrl100, previewUrl, trackId } = item;
  return (
    <CustomCard data-testid="track-card">
      <Link to={`/track/${trackId}`}>
        <ImageCard>
          <Image alt="artwork" src={artworkUrl100} />
          <TextCard data-testid="text-card" title={trackName} description={artistName} />
        </ImageCard>
      </Link>
      <CustomButton>
        <If condition={isPlaying && item.trackId === currentPlayingId}>
          <CustomPause data-testid="pause-button" onClick={() => onPause(previewUrl)} />
        </If>
        <If condition={!(isPlaying && item.trackId === currentPlayingId)}>
          <CustomPlay data-testid="play-button" onClick={() => onPlay(previewUrl, trackId)} />
        </If>
        <CustomRight />
      </CustomButton>
    </CustomCard>
  );
}

TrackCard.propTypes = {
  height: PropTypes.number,
  item: PropTypes.object,
  onPause: PropTypes.func,
  onPlay: PropTypes.func,
  currentPlayingId: PropTypes.number,
  isPlaying: PropTypes.bool,
  trackId: PropTypes.number
};

export default TrackCard;
