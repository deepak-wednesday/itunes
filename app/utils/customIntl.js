import translations from '../translations/en.json';
import { createIntl, createIntlCache } from '@formatjs/intl';

function customIntl() {
  return createIntl(
    {
      locale: 'en',
      messages: translations
    },
    createIntlCache()
  );
}
export default customIntl;
