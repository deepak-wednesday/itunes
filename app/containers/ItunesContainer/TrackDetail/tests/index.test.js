/**
 *
 * Tests for TrackDetail
 *
 */

import React from 'react';
import { TrackDetailTest as TrackDetail } from '../index';
// import { fireEvent } from '@testing-library/dom'
import { timeout, renderWithIntl } from '@utils/testUtils';
import { setIntl } from '@app/components/IntlGlobalProvider/';
import customIntl from '@app/utils/customIntl';

describe('<TrackDetail />', () => {
  let submitSpy, submitSpyClear;
  beforeEach(() => {
    submitSpy = jest.fn();
    submitSpyClear = jest.fn();
  });

  beforeAll(() => {
    setIntl(customIntl());
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(
      <TrackDetail dispatchTrackData={submitSpy} dispatchClearTrackData={submitSpyClear} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should dispatchClearTrackData & dispatchTrackData ', async () => {
    renderWithIntl(<TrackDetail dispatchTrackData={submitSpy} dispatchClearTrackData={submitSpyClear} />);
    await timeout(500);
    expect(submitSpyClear).toBeCalled();
    expect(submitSpy).toBeCalled();
  });
});
