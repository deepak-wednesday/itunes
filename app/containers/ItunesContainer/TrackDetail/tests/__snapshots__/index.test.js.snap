// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetail /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackDetail__Container-sc-1rmz6ii-0 jxLkoL"
    >
      <div
        class="ant-card ant-card-bordered TrackDetail__CustomCard-sc-1rmz6ii-1 ggcAzF"
        data-testid="track-details-card"
        style="width: 300px;"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="example"
            class="TrackDetail__StyledImage-sc-1rmz6ii-3 gByhtW"
            data-testid="track-image"
            src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/100x100bb.jpg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Upside Down
              </div>
              <div
                class="ant-card-meta-description"
              >
                Jack Johnson
              </div>
            </div>
          </div>
          <button
            class="ant-btn ant-btn-circle ant-btn-icon-only TrackDetail__CustomButtons-sc-1rmz6ii-4 RUOfx"
            type="button"
          >
            <span
              aria-label="play-circle"
              class="anticon anticon-play-circle TrackDetail__CustomPlay-sc-1rmz6ii-5 ggaNOj"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="play-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  fill="#1890ff"
                />
                <path
                  d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm164.1 378.2L457.7 677.1a8.02 8.02 0 01-12.7-6.5V353a8 8 0 0112.7-6.5l218.4 158.8a7.9 7.9 0 010 12.9z"
                  fill="#e6f7ff"
                />
                <path
                  d="M676.1 505.3L457.7 346.5A8 8 0 00445 353v317.6a8.02 8.02 0 0012.7 6.5l218.4-158.9a7.9 7.9 0 000-12.9z"
                  fill="#1890ff"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="TrackDetail__ListContainer-sc-1rmz6ii-2 evKjfX"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Jack Johnson and Friends: Sing-A-Longs and Lullabies for the Film Curious George
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                Rock
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Upside Down
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                Jack Johnson and Friends: Sing-A-Longs and Lullabies for the Film Curious George
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Upside Down
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                Rock
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Upside Down
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                Jack Johnson and Friends: Sing-A-Longs and Lullabies for the Film Curious George
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Jack Johnson and Friends: Sing-A-Longs and Lullabies for the Film Curious George
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                Rock
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
          data-testid="list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-image"
              >
                <img
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/ae/4c/d4/ae4cd42a-80a9-d950-16f5-36f01a9e1881/source/30x30bb.jpg"
                />
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Upside Down
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                1.29
              </div>
            </div>
          </div>
        </li>
      </div>
    </div>
  </div>
</body>
`;
